packages/compass-e2e-tests/helpers/insert-data.ts

```diff
@@ -1,168 +1,217 @@
-      addresses: [`${i}-address1`, `${i}-address2`],
-      phoneNumbers: [
-        {
-          label: `${i}-home`,
-          number: `${i}-12345`,
-        },
-        {
-          label: `${i}-work`,
-          number: `${i}-6789`,
-        },
-      ],
-    }))
+  await Promise.all(
+    test_dbs.map(async (db) => {
+      await db.collection(name).insertMany(
+        [...Array(numberOfRecords).keys()].map((i) => ({
+          names: {
+            firstName: `${i}-firstName`,
+            lastName: `${i}-lastName`,
+          },
+          addresses: [`${i}-address1`, `${i}-address2`],
+          phoneNumbers: [
+            {
+              label: `${i}-home`,
+              number: `${i}-12345`,
+            },
+            {
+              label: `${i}-work`,
+              number: `${i}-6789`,
+            },
+          ],
+        }))
+      );
+    })
   );
 }
 
 export async function createNumbersCollection(
   name = 'numbers',
   numberOfRecords = 1000
 ): Promise<void> {
-  const db = client.db('test');
-
-  await db
-    .collection(name)
-    .insertMany([...Array(numberOfRecords).keys()].map((i) => ({ i, j: 0 })));
+  await Promise.all(
+    test_dbs.map(async (db) => {
+      await db
+        .collection(name)
+        .insertMany(
+          [...Array(numberOfRecords).keys()].map((i) => ({ i, j: 0 }))
+        );
+    })
+  );
 }
 
 // Useful for testing collation with `numericOrdering`.
 export async function createNumbersStringCollection(
   name = 'numbers-strings',
   numberOfRecords = 10
 ): Promise<void> {
-  const db = client.db('test');
-
-  await db.collection(name).insertMany(
-    [...Array(numberOfRecords).keys()].map((i) => ({
-      i,
-      iString: `${i * 20}`,
-      j: 0,
-    }))
+  await Promise.all(
+    test_dbs.map(async (db) => {
+      await db.collection(name).insertMany(
+        [...Array(numberOfRecords).keys()].map((i) => ({
+          i,
+          iString: `${i * 20}`,
+          j: 0,
+        }))
+      );
+    })
   );
 }
 
-export async function createMultipleCollections(): Promise<void> {
-  const db = client.db('multiple-collections');
-
-  await db
-    .collection('one')
-    .insertMany([...Array(10).keys()].map((i) => ({ i, j: 0 })));
-
-  await db
-    .collection('two')
-    .insertMany([...Array(10).keys()].map((i) => ({ i, j: 0 })));
-}
-
 export async function createGeospatialCollection(): Promise<void> {
-  const db = client.db('test');
+  await Promise.all(
+    test_dbs.map(async (db) => {
+      const lon = () => Math.random() * 360 - 180;
+      const lat = () => Math.random() * 180 - 90;
+
+      await db.collection('geospatial').insertMany(
+        [...Array(1000).keys()].map(() => ({
+          location: { type: 'Point', coordinates: [lon(), lat()] },
+        }))
+      );
+    })
+  );
+}
 
-  const lon = () => Math.random() * 360 - 180;
-  const lat = () => Math.random() * 180 - 90;
+// WARNING: please don't export _dropDatabase because this file is written to
```

Comment:
Added this warning and renamed the function to have an underscore in front of it because every once in a while someone starts using it on the outside. Also started writing documentation on how to write/debug e2e tests, what our standards and best practises are and so on.

Reply:
Also some more comments at the top of the file around how to use test databases and collections and where to add them.